sips/sip-011/sip-011-catastrophic-failure-recovery.md

```diff
@@ -0,0 +1,464 @@
+that the source code to the fix is available to be built.
+4. A build process will be triggered to produce pre-built binaries for users to
+download.
+5. Once the binaries are available, a cryptographically-signed email will be
+sent from announce@stacks.org with the cryptographic digests of the binaries and
+download links to them.  The email will detail any special considerations and
+instructions for recovering from the bug.
+
+Anyone can subscribe to receive critical bugfix emails by sending an email to
+announce+subscribe@stacks.org.  Critical bugfix announcements will also be
+posted to the Stacks forum (https://forum.stacks.org).
+
+This procedure is further incorporated into procedures described below, and is
+referenced as the "critical bugfix point-release procedure."
+
+## Procedure for Pushing out Sensitive Bugfixes
+
+Depending on the severity of the bug, it may be necessary for ecosystem
+participants to quietly fix and/or reboot their nodes with a bugfix before
+making any announcement.  For example, if the bug is an as-of-yet-unexploited
+safety failure that could lead to loss or theft of digital assets, it may be
+more prudent to silently upgrade vulnerable nodes before announcing the
+vulnerability.
+
+If it is necessary to do this, then the Foundation or an agent of its choosing
+will reach out to vulnerable ecosystem participants with instructions on how to
+proceed.  In order to minimize the chance that the details of an unexploited but
+critical bug will be leaked, the Foundation (or its agent) may take any means
+necessary to ensure its secrecy in order to protect unknowingly-vulnerable
+users.  This may include, but is not limited to, requiring ecosystem
+participants so contacted to sign a legally-binding non-disclosure agreement.
+
+Once the vulnerability no longer poses a systemic threat to the Stacks ecosystem
+(or once the bug gets exposed, whichever happens first), the procedure for
+pushing out a critical bugfix point-release will be followed.
+
+To encourage users who discover such sensitive blockchain bugs to report them
+while keeping them secret, the Stacks Foundation will start a bug-bounty program that
+will be set up once this SIP activates.
+
+## Procedure for Altering Network Protocols
+
+If a catastrophic failure can be traced to a bug in the Stacks blockchain's
+network protocols (see SIP-003), and if it can be fixed without changing the
+protocol(s) in an incompatible way, then it will be fixed through the critical
+bugfix point-release procedure above.
+
+The Stacks network protocols offer nodes a lot of leeway in how they are allowed
+to behave, so it is unlikely that the protocol itself will need to be altered in
+an incompatible way in order to recover from a catastrophic failure.  But that
+does not by itself preclude the possibility of a catastrophic failure
+originating from the network's design and/or implementation.  In all likelihood,
```

Comment:
We could point to the recent bug with block limits, or the Atlas issues as examples?